### DISCUSSION TITLE
Remove app from sale using fastlane

### DISCUSSION THREAD

**Discussion by Jadoking** [fastlane/fastlane#19040] [Q&A]
Hi I was wondering if it is possible to remove an app from sale using fastlane?

Currently I'm using 

```
app = Spaceship::ConnectAPI::App.find(options[:bundle_id])
app.update(attributes:{"availableInNewTerritories": false},app_price_tier_id:"0",territory_ids:[])
```
I've also tried setting `territory_ids` to `nil` it seems that it just maintains it's previous state. 

Thank you

**Comment by lucgrabowski** [comment]
Hi @jeisun-nagooen, I've removed previous comment as my wild guess wasn't a good one. To be able to remove app from sale the way you proposed above (and it looks like a proper way to do it), a change in fastlane code is needed. It's not possible to use empty array for `territory_ids` currently in app update request. I'll open a pull request to fix this.

**Comment by lucgrabowski** [comment]
I've opened a PR to allow for removing an app from sale: https://github.com/fastlane/fastlane/pull/20324
